### ISSUE TITLE
binary search should be an assoc

### ISSUE THREAD

**Issue by lolbinarycat** [factor/factor#2341]
this would make it easier to slot into existing systems, and would reduce the need for specific words.

**Comment by mrjbq7** [comment]
You mean ``binary-search`` should work on an assoc?

**Comment by lolbinarycat** [comment]
No, I mean that there should be a binary-search (or perhaps bsearch) class, which would serve as a marker that the contents were sorted.
